examples/permissioned-token/test/failing-paths.js

```diff
@@ -0,0 +1,178 @@
+      appArgs: ['str:add_whitelist', `int:${controllerAppID}`],
+      accounts: [address],
+      foreignAssets: [assetIndex],
+      foreignApps: [controllerAppID]
+    }), 'RUNTIME_ERR1009');
+  });
+
+  it('Opt Out', () => {
+    // Opt-out wont work without opting in
+    assert.throws(() => optOut(runtime, alice, elon, assetIndex), 'RUNTIME_ERR1404');
+  });
+
+  it('Change Permissions SSC Manager', () => {
+    // Optin In new SSC manager to the SSC
+    optInToPermissions(runtime, elon.address, permissionsAppId);
+    syncAccounts();
+
+    // whitelist operation can only be performed by permissions SSC manager
+    assert.throws(() => whitelist(runtime, elon, bob.address, assetIndex, controllerAppID, permissionsAppId), 'RUNTIME_ERR1009');
+
+    const txn = {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: bob.account, // asset manager account (fails otherwise)
+      appId: permissionsAppId,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:change_permissions_manager'],
+      accounts: [elon.address]
+    };
+
+    // fails as fromAccount is not the current SSC manager
+    assert.throws(() => runtime.executeTx(txn), 'RUNTIME_ERR1009');
+  });
+
+  it('Force Transfer', () => {
+    // Opt-In to permissions SSC & Whitelist
+    whitelist(runtime, alice, alice.address, assetIndex, controllerAppID, permissionsAppId);
+    whitelist(runtime, alice, elon.address, assetIndex, controllerAppID, permissionsAppId);
+    whitelist(runtime, alice, bob.address, assetIndex, controllerAppID, permissionsAppId);
+    syncAccounts();
+
+    // Opt-In to ASA
+    optInToASA(runtime, bob.address, assetIndex);
+    optInToASA(runtime, elon.address, assetIndex);
+    syncAccounts();
+
+    // Issue some tokens to sender
+    issue(runtime, alice, bob, 150, controllerAppID, assetIndex, lsig);
+    syncAccounts();
+
+    // Fails as only manager can perform force transfer
+    assert.throws(() =>
+      forceTransfer(runtime, bob, elon, 20, assetIndex, controllerAppID, permissionsAppId, lsig, bob),
+    'RUNTIME_ERR1009'
+    );
+
+    // Fails as reciever will have more than 100 tokens post succesfull transaction
+    assert.throws(() =>
+      forceTransfer(runtime, bob, elon, 101, assetIndex, controllerAppID, permissionsAppId, lsig, alice),
+    'RUNTIME_ERR1009'
+    );
+  });
+
+  it('Token Transfer', () => {
+    fund(runtime, master, elon.address);
+    fund(runtime, master, bob.address);
+    fund(runtime, master, alice.address);
+    fund(runtime, master, lsig.address());
+    syncAccounts();
+    const amount = 20;
+
+    // Issue some tokens to Bob and Elon
+    optInToASA(runtime, elon.address, assetIndex);
+    optInToASA(runtime, bob.address, assetIndex);
+    syncAccounts();
+    issue(runtime, alice, elon, 50, controllerAppID, assetIndex, lsig);
+    issue(runtime, alice, bob, 50, controllerAppID, assetIndex, lsig);
+    syncAccounts();
+
+    // Cannot transfer directly
```

Comment:
What do we mean by "Cannot transfer directly"?

Reply:
Direct ASA transfer. We can only transfer by making a clawback transaction.